pennylane/measurements/mid_measure.py

```diff
@@ -78,8 +80,8 @@ def func(x, y):
 
     # Create a UUID and a map between MP and MV to support serialization
     measurement_id = str(uuid.uuid4())[:8]
-    MidMeasureMP(wires=wire, id=measurement_id)
-    return MeasurementValue([measurement_id], processing_fn=lambda v: v)
+    mp = MidMeasureMP(wires=wire, id=measurement_id)
+    return MeasurementValue([mp], processing_fn=lambda v: v)
```

Comment:
seems like we won't really need `measurement_id` after this change. Can we just delete it, and just use the default `None` in `MidMeasureMP`?

Reply:
The id is still useful for sorting (see comment below)